spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanParameterResolverFactory.java

```diff
@@ -76,16 +79,24 @@ public ParameterResolver createInstance(Executable executable, Parameter[] param
         } else if (beansFound.size() > 1) {
             final AutowireCapableBeanFactory beanFactory = applicationContext.getAutowireCapableBeanFactory();
             if (beanFactory instanceof ConfigurableListableBeanFactory) {
+                final ConfigurableListableBeanFactory clBeanFactory = (ConfigurableListableBeanFactory) beanFactory;
+                // find @Qualifier matching candidate
+                for (Map.Entry<String, ?> bean : beansFound.entrySet()) {
+                    final Qualifier qualifier = parameters[parameterIndex].getAnnotation(Qualifier.class);
```

Comment:
The `qualifier` variable can be taken outside the `for` scope, as it doesn't depend on that. Also don't go into the `for`loop when the `qualifier` is `null`.

Lastly, please use `org.axonframework.common.annotation.AnnotationUtils.findAnnotationAttributes(...)` to retrieve annotations, because it takes meta-annotations into account. The key you'd want to look for in the returned Map is 'qualifier'.